src/test/scala/it/pagopa/interop/attributeregistryprocess/SpecHelper.scala

```diff
@@ -0,0 +1,90 @@
+package it.pagopa.interop.attributeregistryprocess
+
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import com.nimbusds.jwt.JWTClaimsSet
+import com.typesafe.config.{Config, ConfigFactory}
+import it.pagopa.interop.attributeregistrymanagement.client.{model => ManagementDependency}
+import it.pagopa.interop.attributeregistryprocess.api.AttributeApiService
+import it.pagopa.interop.attributeregistryprocess.{model => ProcessDependency}
+import it.pagopa.interop.attributeregistryprocess.api.impl.AttributeRegistryApiServiceImpl
+import it.pagopa.interop.attributeregistryprocess.service.{AttributeRegistryManagementService, PartyRegistryService}
+import it.pagopa.interop.commons.cqrs.service.ReadModelService
+import it.pagopa.interop.commons.utils.service.{OffsetDateTimeSupplier, UUIDSupplier}
+import it.pagopa.interop.partyregistryproxy.client.model.{Categories, Institutions}
+import org.mongodb.scala.bson.conversions.Bson
+import org.scalamock.scalatest.MockFactory
+import spray.json._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Success, Try}
+
+trait SpecHelper extends SprayJsonSupport with DefaultJsonProtocol with MockFactory {
+
+  final val bearerToken: String = "token"
+
+  val config: Config = ConfigFactory
+    .parseResourcesAnySyntax("application-test")
+    .resolve()
+
+  val mockReadModel: ReadModelService                                            = mock[ReadModelService]
+  val mockAttributeRegistryManagementService: AttributeRegistryManagementService =
+    mock[AttributeRegistryManagementService]
+  val mockPartyRegistryService: PartyRegistryService                             = mock[PartyRegistryService]
+
+  val mockUUIDSupplier: UUIDSupplier               = mock[UUIDSupplier]
+  val mockDateTimeSupplier: OffsetDateTimeSupplier = mock[OffsetDateTimeSupplier]
+
+  val service: AttributeApiService = AttributeRegistryApiServiceImpl(
+    mockAttributeRegistryManagementService,
+    mockUUIDSupplier,
+    mockDateTimeSupplier,
+    mockPartyRegistryService,
+    mockReadModel
+  )(ExecutionContext.global)
+
+  def mockSubject(uuid: String): Try[JWTClaimsSet] = Success(new JWTClaimsSet.Builder().subject(uuid).build())
+
+  def delta(
+    attrs: List[ProcessDependency.Attribute],
+    attributesSeeds: Seq[ProcessDependency.AttributeSeed]
+  ): Set[ProcessDependency.AttributeSeed] =
+    attributesSeeds.foldLeft[Set[ProcessDependency.AttributeSeed]](Set.empty)((attributesDelta, seed) =>
+      attrs
+        .find(persisted => seed.origin == persisted.origin && seed.code == persisted.code)
+        .fold(attributesDelta + seed)(_ => attributesDelta)
+    )
```

Comment:
we are testing this logic with the logic itself, if this contains a bug we would not be able to see it.
I suggest to just manually create the objects you use as input and expect as output.
It is more verbose, but it is also more clear and solid.
```suggestion
```